### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds the GitHub action [PR Labeler](https://github.com/actions/labeler) to auto-label PRs based on their content. 

Labeling is managed with a configuration file `.github/labeler.yml` using the following [options](https://github.com/actions/labeler#usage).

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  
+libclx:
+  - 'cpp/**'
+
+CMake:
+  - '**/CMakeLists.txt'
+  - '**/cmake/**'
+  
+Java:
+  - 'java/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:17]: This section should match this repo's labels and the contents of the .github/CODEOWNERS file as a starting guide - PICs in this repo would then add or suggest more labels as needed

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  
+Ops:
+  - '.github/**'
+  - 'ci/**'
+  - 'conda/**'
+  - '**/Dockerfile'
+  - '**/.dockerignore'
+  - 'docker/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:25]: This should be removed

**Review by mike-wendt — CHANGES_REQUESTED** [review]

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# https://github.com/actions/labeler#common-examples
+# Adapted from https://github.com/rapidsai/clx/blob/main/.github/CODEOWNERS
+# Labels culled from https://github.com/rapidsai/clx/labels
+
+Python:
+  - 'python/**'
+  - 'notebooks/**'
+
+ci:
```
Comment (ajschmidt8) [inline .github/labeler.yml:9]:
```suggestion
gpuCI:
```
there's no `ci` label, so this can probably be changed to `gpuCI`.

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Comment by mike-wendt** [comment]
@BartleyR any other labels based on content you would like us to add here? Please add your review when you can

**Review by BartleyR — CHANGES_REQUESTED** [review]
I think it'd be good to have an `integration` label that we could use to label things in the `siem_integrations` folder for now. We would also be able to use it to label future integrations. I went ahead and made the label for the repo.

**Comment by mike-wendt** [comment]
> I think it'd be good to have an `integration` label that we could use to label things in the `siem_integrations` folder for now. We would also be able to use it to label future integrations. I went ahead and made the label for the repo.

@jolorunyomi can you add this to the config? thanks!

**Review by mike-wendt — APPROVED** [review]

**Comment by mike-wendt** [comment]
@BartleyR I think we're ready for a review, thanks for the help

**Review by BartleyR — APPROVED** [review]

**Comment by mike-wendt** [comment]
@gpucibot merge

**Pull request merged**